MinStack.java

```diff
@@ -0,0 +1,49 @@
+// Time Complexity O(1)
+// Space Complexity O(1)
+
```

Comment:
Please check the Space Complexity. We are storing `min`, which in worse case can be O(n) where n is the number of items in the stack.